### ISSUE TITLE
Accept custom Rpc

### ISSUE THREAD

**Issue by adeojoemmanuel** [solana-labs/solana-web3.js#2460]
## Motivation

so i was getting a rate limit on the devnet cluter, and had to use a third party platform which helped, but the issue was i couldnt pass in this custom RPC into the clusterApiUrl method.

## Example use case

i get a custom RPC, probably a paid provider,  and want to pass in the url into the clusterApiUrl method so as to validate the custom RPC and certify it meet standard 

## code sample

  ```
const customRpcUrl = 'https://customRpcCluter.net';

const connection = new Connection(clusterApiUrl(customRpcUrl), { commitment: "confirmed" });
```

## Details

  basically accepting a custom rpc into the clusterApiUrl method, which helps to validate the custom RPC, at the moment the types available are 

`type Cluster = 'devnet' | 'testnet' | 'mainnet-beta';`

so if a new type is added called custom, which accept a url, validate the url and allow of

**Comment by steveluscher** [comment]
That function is only for getting the default RPC urls given a cluster.

https://github.com/solana-labs/solana-web3.js/blob/18d6b56a69509e4c98de8f3de51abe2623b46763/packages/library-legacy/src/utils/cluster.ts#L16-L22

To use your own, simply supply the URL as the first parameter.
